src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,61 @@
+package duke.task;
+
+import duke.formatter.DateFormatter;
+import duke.parser.DateParser;
+import java.util.Date;
+
+/**
+ * A deadline item is a task with description and the date the deadline task is due by. The deadline item can be
+ * created, marked as done or deleted.
+ */
+public class Deadline extends Task {
+
+    /**
+     * This is the date where the deadline task is due by.
+     */
+    private Date by;
+
+    /**
+     * Constructs a new deadline task with description and the date where the deadline task is due by that has not been
+     * done.
+     * @param description the description of the deadline task
+     * @param by the date the task is due by
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = DateParser.parse(by);
```

Comment:
An alternative way of implementation is to keep the type conversion inside the `Parser` class. Hence, the constructor can take `Date` type as parameter directly. The benefit is that all type conversion related exceptions can be caught in the `Parser` class instead of being thrown by constructors of different classes.